packages/flutter/lib/src/widgets/form.dart

```diff
@@ -697,6 +700,20 @@ class FormFieldState<T> extends State<FormField<T>> with RestorationMixin {
     }
   }
 
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (Form.maybeOf(context)?.widget.autovalidateMode == AutovalidateMode.always) {
+      if (!_buildScheduled) {
+        _buildScheduled = true;
+        SchedulerBinding.instance.addPostFrameCallback((_) {
+          _buildScheduled = false;
+          validate();
+        });
+      }
+    }
+  }
```

Comment:
Looks like a super effective solution, especially after seeing the test you added!

<br>

I noticed that the FormState's `_register()` method currently violates a style guideline:

```dart
  void _register(FormFieldState<dynamic> field) {
    _fields.add(field);
    if (widget.autovalidateMode == AutovalidateMode.onUnfocus) {
      field._focusNode.addListener(() => _updateField(field));
    }
  }
```

I'm wondering… maybe turning this into a switch statement and adding logic for `AutovalidateMode.always` could allow solving this issue without any post-frame callbacks.

Let me know what you think 🙂

Reply:
The switch statement worked beautifully and also made the code cleaner and more straightforward. Which guideline is `_register()` violating, those that were recently added? I'll take the opportunity to apply that improvement as well.

Reply:
There is one issue: the Material 2 time picker is not working as expected. I'm trying to figure out the issue, even with adding WidgetsBinding.instance.addPostFrameCallback(...) it still doesn't work. 😔😔

Reply:
Dang, unfortunate to hear.

<br>

As far as the style guideline, I was going to link to it but forgot, my bad!

https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-if-chains-or--or--with-enum-values

> ### Avoid using `if` chains or `?:` or `==` with enum values
> Use `switch` with no `default` case if you are examining an enum, since the analyzer will warn you if you missed any of the values when you use `switch`. The `default` case should be avoided so that the analyzer will complain if a value is missing. Unused values can be grouped together with a single `break` or `return` as appropriate.

Reply:
I found the problem, for example in the tests [time_picker_theme_test](https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/time_picker_theme_test.dart#L862-L896) the `await tester.pumpAndSettle();` causes the problem of multiple builds. If I remove them, it works normally.

My code looks like this:
```
void _register(FormFieldState<dynamic> field) {
    _fields.add(field);
    switch (widget.autovalidateMode) {
      case AutovalidateMode.always:
        if (field.widget.enabled) {
          field.validate();
        }
      case AutovalidateMode.onUnfocus:
        field._focusNode.addListener(() => _updateField(field));
      case AutovalidateMode.onUserInteraction:
      case AutovalidateMode.disabled:
        break;
    }
  }
```
And if I add:

```
WidgetsBinding.instance.addPostFrameCallback((_) {
  if (field.widget.enabled) {
    field.validate();
  }
});
```
It causes a memory leak, because of `await tester.pumpAndSettle()`;

Reply:
I resolved the issue by simply moving the logic from inside the `Form` to within the `FormField`.